### ISSUE TITLE
3.5.0-rc1: Bad PROMPT_COMMAND gives (user) unintelligible error messages.

### ISSUE THREAD

**Issue by pontus** [apptainer/singularity#4690]
### Version of Singularity:

What version of Singularity are you using? Run:

```
[pontusf@snowy-lr2 ~]$ singularity version
3.5.0~rc.1-1.el7
[pontusf@snowy-lr2 ~]$ 
```

### Expected behavior

What did you expect to see when you do...?

```
singularity shell  test.sif                  
```

would be a prompt

```
Singularity>
```

### Actual behavior

What actually happend? Why was it incorrect?

```
[pontusf@snowy-lr2 ~]$ singularity shell test.sif
bash: 172.18.84.2: command not found
Singularity> 
```

### Steps to reproduce this behavior

This seems caused by us running with a special PROMPT_COMMAND:

```
[pontusf@snowy-lr2 ~]$ echo "$PROMPT_COMMAND"
_siginttrap=`trap -p SIGINT`; trap  SIGINT; history -a; history -n; echo -n -e "\\033k${USER}@${hname}\033\\";history -a >(tee -a ~/.bash_history | logger -t "$USER[$$] $SSH_CONNECTION"); eval "$_siginttrap"
[pontusf@snowy-lr2 ~]$ 
```
This seems to cause something to break with the new prompt handling, inside the container:

```
                                                                                                   
[pontusf@snowy-lr2 ~]$ singularity shell test.sif
bash: 172.18.84.2: command not found
Singularity> echo $PROMPT_COMMAND
_siginttrap=`trap -p SIGINT`; trap SIGINT; history -a; history -n; echo -n -e "\\033k${USER}@${hname}\033\\";history -a >(tee -a ~/.bash_history | logger -t "$USER[$$] $SSH_CONNECTION"); eval "$_siginttrap"; PS1="Singularity> "; PROMPT_COMMAND="_siginttrap=`trap -p SIGINT`; trap SIGINT; history -a; history -n; echo -n -e "\\033k${USER}@${hname}\033\\";history -a >(tee -a ~/.bash_history | logger -t "$USER[$$] $SSH_CONNECTION"); eval "$_siginttrap""
bash: 172.18.84.2: command not found
Singularity> 
```

I'll try to have a look and figure out if I can see why, but figured I should try to report this first.

How can others reproduce this issue/problem?

### What OS/distro are you running

```
NAME="CentOS Linux"
VERSION="7 (Core)"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="7"
PRETTY_NAME="CentOS Linux 7 (Core)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:centos:centos:7"
HOME_URL="https://www.centos.org/"
BUG_REPORT_URL="https://bugs.centos.org/"

CENTOS_MANTISBT_PROJECT="CentOS-7"
CENTOS_MANTISBT_PROJECT_VERSION="7"
REDHAT_SUPPORT_PRODUCT="centos"
REDHAT_SUPPORT_PRODUCT_VERSION="7"
```

### How did you install Singularity

Custom built RPM (sudo -E rpmbuild -tb singularity-3.5.0-rc.1.tar.gz)

**Comment by pontus** [comment]
A simpler PROMPT_COMMAND set in some cases seem not to have this issue:

```
_siginttrap=$(trap -p SIGINT); trap SIGINT; history -a >(tee -a ~/.bash_history | logger -p local5.info -t "$USER[$$] $SSH_CONNECTION"); eval "$_siginttrap"

```

**Comment by cclerget** [comment]
Hi @pontus , #4692 should fix that if you can give it a try that would much appreciated. Thanks !

**Comment by pontus** [comment]
Tested by cut-n-paste to /etc/singularity/actions/FILE, and it seems to work for all three cases, thanks a lot.

**Comment by dtrudg** [comment]
Thanks for reporting and testing @pontus - merged for the impending `3.5.0-rc.2` which I'll put out today or tomorrow.
